<launch>
  <arg name="command_input" default="2" />
  <arg name="gazebo_simulation" default="true" />
  <arg name="visualization" default="true"/>
  <arg name="log_output" default="screen" />
  <arg name="max_acc" default="2.5" />
  <arg name="max_vel" default="1.5" />
  <arg name="takeoff_height" default="2.0" />
  
  <node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
  		<!-- <param name="mav_name" type="string" value="$(arg mav_name)" /> -->
          <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
          <param name="ctrl_mode" value="$(arg command_input)" />
          <param name="enable_sim" value="$(arg gazebo_simulation)" />
          <param name="enable_gazebo_state" value="true"/>
          <param name="posehistory_window" value="0" />
          <!-- <param name="max_acc" value="10.0" /> -->
          <param name="max_acc" value="8.0" />
          <param name="takeoff_height" value="$(arg takeoff_height)" />
          <!--Params specific for Iris copter-->
          <param name="attctrl_constant" value="0.3"/>
          <param name="normalizedthrust_constant" value="0.06"/>
          <param name="normalizedthrust_offset" value="0.1"/>
          <param name="Kp_x" value="10.0" />
          <param name="Kp_y" value="10.0" />
          <param name="Kp_z" value="20.0" />
          <param name="Kv_x" value="5.0"/>
          <param name="Kv_y" value="5.0"/>
          <param name="Kv_z" value="10.0"/>
  </node>
  
  <!-- <node pkg="opendrone" type="test1" name="test" output="screen">
        <param name="max_acc" value="$(arg max_acc)" />
  </node> -->

  <!--- Trajectory planner -->
  <node pkg="opendrone" type="planner_node" name="planner_node" output="screen">
        <param name="max_acc" value="$(arg max_acc)" />
        <param name="max_vel" value="$(arg max_vel)" />
        <remap from="/trajectory_generation/waypoint" to="/waypoint_generator/waypoints" />
  </node>

  <!--- Trajectory sampler -->
  <node name="sampler" pkg="mav_trajectory_generation_ros" type="trajectory_sampler_node" output="screen">
        <param name="publish_whole_trajectory" value="false" />
        <remap from="path_segments_4D" to="/trajectory_generation/trajectory"/>
  </node>

  <node pkg="waypoint_generator" type="waypoint_generator" name="waypoint_generator" output="screen">
        <param name="waypoint_type" value="circle"/>
        <remap from="~odom" to="/mavros/local_position/odom"/>
  </node>

  <!-- Launch rqt_reconfigure -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" name="rqt_reconfigure" />

  <!-- <group if="$(arg visualization)">
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find geometric_controller)/launch/config_file.rviz" />
  </group> -->

</launch>